### ISSUE TITLE
Building project results in compilation error with Xcode 14 beta IDE

### ISSUE THREAD

**Issue by JoshChang-TomTom** [boostorg/boost#688]
Apple announced the new Xcode 14 beta IDE.
It causes 3 build errors in dependent Boost lib which uses deprecated sprintf:

> Boost/headers/lexical_cast/detail/converter_lexical_streams.hpp
> 'sprintf' is deprecated: This function is provided for compatibility reasons only. Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.

Extra information:
Boost version used: 1_75_0 (The latest version 1_80_0 has the same issue)

Use Xcode 14 Compiler:
> Apple clang version 14.0.0 (clang-1400.0.28.1)

On previous Xcode 13, it worked well.
Xcode 13 Compiler:
> Apple clang version 13.1.6 (clang-1316.0.21.2.5)

**Comment by mclow** [comment]
Warning, not error.  But still...

**Comment by cpsauer** [comment]
Indeed, though not obvious to users that there isn't a real security issue here?
I know that sprintf calls are also present in other places, like regex

**Comment by Hadatko** [comment]
Hello, isn't better to replace sprintf with snprintf? Using '-Wno-deptecated-declarations' you are forcing other projects to use that too. For example we are threating to all warnings as errors.

`/opt/homebrew/Cellar/boost/1.81.0_1/include/boost/lexical_cast/detail/converter_lexical_streams.hpp:310:21: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
`
https://github.com/EmbeddedRPC/erpc/issues/342

**Comment by mclow** [comment]
In lexical_cast, this has been resolved (see https://github.com/boostorg/lexical_cast/pull/56). It will be in the next boost release.

**Comment by Hadatko** [comment]
Oh great, 👍👍👍
